conans/test/functional/graph/test_graph_build_mode.py

```diff
@@ -0,0 +1,167 @@
+                .with_require("bar/1.0@user/testing")
+                .with_setting("build_type")})
+    client.run("export . foobar/1.0@user/testing")
+    client.run("install foobar/1.0@user/testing --build")
+
+    return client
+
+
+def test_install_build_single(build_all):
+    """ When only --build=<ref> is passed, only <ref> must be built
+    """
+    build_all.run("install foobar/1.0@user/testing --build=foo")
+
+    assert "bar/1.0@user/testing:7839863d5a059fc6579f28026763e1021268c55e - Cache" in build_all.out
+    assert "foo/1.0@user/testing:4024617540c4f240a6a5e8911b0de9ef38a11a72 - Build" in build_all.out
+    assert "foobar/1.0@user/testing:89636fbae346e3983af2dd63f2c5246505e74be7 - Cache" in build_all.out
+    assert "foo/1.0@user/testing: Forced build from source" in build_all.out
+    assert "bar/1.0@user/testing: Forced build from source" not in build_all.out
+    assert "foobar/1.0@user/testing: Forced build from source" not in build_all.out
+    assert "No package matching" not in build_all.out
+
+
+def test_install_build_double(build_all):
+    """ When both --build=<ref1> and --build=<ref2> are passed, only both should be built
+    """
+    build_all.run("install foobar/1.0@user/testing --build=foo --build=bar")
+
+    assert "bar/1.0@user/testing:7839863d5a059fc6579f28026763e1021268c55e - Build" in build_all.out
+    assert "foo/1.0@user/testing:4024617540c4f240a6a5e8911b0de9ef38a11a72 - Build" in build_all.out
+    assert "foobar/1.0@user/testing:89636fbae346e3983af2dd63f2c5246505e74be7 - Cache" in build_all.out
+    assert "foo/1.0@user/testing: Forced build from source" in build_all.out
+    assert "bar/1.0@user/testing: Forced build from source" in build_all.out
+    assert "foobar/1.0@user/testing: Forced build from source" not in build_all.out
+    assert "No package matching" not in build_all.out
+
+
+@pytest.mark.parametrize("build_arg,mode", [("--build", "Build"),
+                                            ("--build=", "Cache"),
+                                            ("--build=*", "Build")])
+def test_install_build_only(build_arg, mode, build_all):
+    """ When only --build is passed, all packages must be built from sources
+        When only --build= is passed, it's considered an error
+        When only --build=* is passed, all packages must be built from sources
+    """
+    build_all.run("install foobar/1.0@user/testing {}".format(build_arg))
+
+    assert "bar/1.0@user/testing:7839863d5a059fc6579f28026763e1021268c55e - {}".format(mode) in build_all.out
+    assert "foo/1.0@user/testing:4024617540c4f240a6a5e8911b0de9ef38a11a72 - {}".format(mode) in build_all.out
+    assert "foobar/1.0@user/testing:89636fbae346e3983af2dd63f2c5246505e74be7 - {}".format(mode) in build_all.out
+    opin = fakeop(lambda x, y: x in y) if mode == "Build" else fakeop(lambda x, y: x not in y)
+    assert "foo/1.0@user/testing: Forced build from source" |opin| build_all.out
```

Comment:
Same as above? please explain what problem you are trying to solve.

Reply:
Regression test. I don't want to break the previous behavior adding a new feature. But we don't have tests which explore `conan create --build` enough to validate it. I could remove, but if this feature breaks anything related to --build, we won't know.

Reply:
Ok, sounds good.